app/models/course.rb

```diff
@@ -167,6 +167,16 @@ def duplication_manifest
       *forums
     ]
   end
+  
+  # Course setting to enable public test cases output
+  def enable_public_test_cases_output
+    settings.enable_public_test_cases_output
+  end
+
+  def enable_public_test_cases_output=(option)
```

Comment:
It's better to move these methods under the specific component (in this case assessment).  You can check how announcements handles settings.

Reply:
Discussed with @anirudhnigania, realised that there's no settings adapter for assessment yet.  

But minimally, we should make sure the settings are stored under the correct component subtree.  The structure will be sth like ```settings: {"course_assessments_component"=>{"settings_key"=>"xxxx", "settings_key_2"=>"xxx"}}```